### DISCUSSION TITLE
Just switched to agile. Rates aren't updating?

### DISCUSSION THREAD

**Discussion by JRascagneres** [BottlecapDave/HomeAssistant-OctopusEnergy#309] [Q&A]
Octopus dashboard shows me as switched but current and next rates are incorrect in the integration. Still shows the variable tarrif rates... Any ideas if I need to do something to get it to update? The standing charge has updated correctly.

**Comment by JRascagneres** [comment]
Just updated home assistant (which restarts HA) and that fixed it... Perhaps there is an enhancement to make this work without a restart? Or it may have just been a coincidence

**Reply by BottlecapDave** [reply]
Hi there. The logic in the background updater should update the rates when they are out of date or every 30 minutes. If rates can't be retrieved, then it will use the cached rates. The only thing I can think of is that the rates failed to retrieve on the 30 minute mark and so had to wait for the next instance. Or that HA is not running the job as frequently as I've requested. I'll have a look at making this a bit more robust.

**Reply by JRascagneres** [reply]
I actually switched at about 1am. So I gave plenty of time
